meson.build

```diff
@@ -79,15 +79,25 @@ endif
 if xapian_dep.found()
     icu_dep = dependency('icu-i18n', static:static_linkage)
     if icu_dep.version().version_compare('>= 76')
-        icu_dep = [icu_dep, dependency('icu-uc', static:static_linkage)]
+        icu_deps = [icu_dep, dependency('icu-uc', static:static_linkage)]
+    else
+        icu_deps = [icu_dep]
     endif
 else
     icu_dep = dependency('icu-i18n', 'required:false', static:static_linkage)
     if icu_dep.version().version_compare('>= 76')
-        icu_dep = [icu_dep, dependency('icu-uc', 'required:false', static:static_linkage)]
+        icu_deps = [icu_dep, dependency('icu-uc', 'required:false', static:static_linkage)]
+    else
+        icu_deps = [icu_dep]
     endif
 endif
```

Comment:
I think that we can avoid the duplication between the positive and negative branches of this `if`, by introducing a variable `icu_dep_is_required = xapian_dep.found()`